tests/lib/rules/no-useless-escape.js

```diff
@@ -127,163 +127,851 @@ ruleTester.run("no-useless-escape", rule, {
     ],
 
     invalid: [
-        { code: "var foo = /\\#/;", errors: [{ line: 1, column: 12, endColumn: 13, message: "Unnecessary escape character: \\#.", type: "Literal" }] },
-        { code: "var foo = /\\;/;", errors: [{ line: 1, column: 12, endColumn: 13, message: "Unnecessary escape character: \\;.", type: "Literal" }] },
-        { code: "var foo = \"\\'\";", errors: [{ line: 1, column: 12, endColumn: 13, message: "Unnecessary escape character: \\'.", type: "Literal" }] },
-        { code: "var foo = \"\\#/\";", errors: [{ line: 1, column: 12, endColumn: 13, message: "Unnecessary escape character: \\#.", type: "Literal" }] },
-        { code: "var foo = \"\\a\"", errors: [{ line: 1, column: 12, endColumn: 13, message: "Unnecessary escape character: \\a.", type: "Literal" }] },
-        { code: "var foo = \"\\B\";", errors: [{ line: 1, column: 12, endColumn: 13, message: "Unnecessary escape character: \\B.", type: "Literal" }] },
-        { code: "var foo = \"\\@\";", errors: [{ line: 1, column: 12, endColumn: 13, message: "Unnecessary escape character: \\@.", type: "Literal" }] },
-        { code: "var foo = \"foo \\a bar\";", errors: [{ line: 1, column: 16, endColumn: 17, message: "Unnecessary escape character: \\a.", type: "Literal" }] },
-        { code: "var foo = '\\\"';", errors: [{ line: 1, column: 12, endColumn: 13, message: "Unnecessary escape character: \\\".", type: "Literal" }] },
-        { code: "var foo = '\\#';", errors: [{ line: 1, column: 12, endColumn: 13, message: "Unnecessary escape character: \\#.", type: "Literal" }] },
-        { code: "var foo = '\\$';", errors: [{ line: 1, column: 12, endColumn: 13, message: "Unnecessary escape character: \\$.", type: "Literal" }] },
-        { code: "var foo = '\\p';", errors: [{ line: 1, column: 12, endColumn: 13, message: "Unnecessary escape character: \\p.", type: "Literal" }] },
+        {
+            code: "var foo = /\\#/;",
+            errors: [{
+                line: 1,
+                column: 12,
+                endColumn: 13,
+                message: "Unnecessary escape character: \\#.",
+                type: "Literal",
+                suggestions: [{
+                    messageId: "removeEscape",
+                    output: "var foo = /#/;"
```

Comment:
> I'm kind of concerned about the tests. Why wouldn't we want to use text-based output and just run the autofixer against the suggestion's fix object? It would be much easier for users to test and it would be consistent with how autofix is tested outside of suggestions.
>
> If the implementation has switched recently to use a text-base approach, that's great (and please see my comment about not having time for open source 😅). Otherwise, I guess I would like to understand what benefit we get from requiring users to know the structure of the fix object (and effectively making it a public API) to write tests for suggestions.

@platinumazure, I have set up support for testing the output of what applying the suggestion would be and that is what I'm doing in these tests (as seen above). As noted [here](https://github.com/eslint/eslint/pull/12384/files#r346592356), this PR also currently supports testing the shape of the suggestion fix object. If you think we should only provide the `output`-based testing option, and not have the ability for users to test the `fix` object I'm fine with removing that option.

Thanks for commenting even though you have limited time. Let me know if there is anything else I can try to clarify.

Reply:
Oh, I just saw your edited comment above. That seems reasonable. I will go ahead and remove the option to test the shape of the fix object.

Reply:
Thanks so much! (I wasn't sure whether to edit or leave a second comment.)

Hoping to have time to do a full review this weekend.